frontend/packages/shared/src/components/GiteaHeader/ThreeDotsMenu/LocalChangesModal/DeleteModal/DeleteModal.test.tsx

```diff
@@ -55,14 +64,59 @@ describe('DeleteModal', () => {
     const textfield = screen.getByLabelText(
       textMock('local_changes.modal_delete_modal_textfield_label'),
     );
-    await act(() => user.type(textfield, mockAppName));
+    await act(() => user.type(textfield, mockApp));
+    expect(deleteButton).not.toBeDisabled();
+
+    expect(mockDelete).toHaveBeenCalledTimes(0);
+    await act(() => user.click(deleteButton));
+    expect(mockDelete).toHaveBeenCalledTimes(1);
+
+    const toastSuccessText = await screen.findByText(
+      textMock('local_changes.modal_deleted_success'),
+    );
+    expect(toastSuccessText).toBeInTheDocument();
+    expect(mockOnClose).toHaveBeenCalledTimes(1);
+  });
 
-    const deleteButtonAfterTypedInName = screen.getByRole('button', {
+  it('does not call the onClose function when the Delete button is clicked and an error is received', async () => {
+    const user = userEvent.setup();
+
+    const mockDelete = jest.fn().mockImplementation(() => Promise.reject());
+
+    render(null, { resetRepoChanges: mockDelete });
```

Comment:
Should we change the render method to take an object as argument, then we do not need to pass `null`. Should we rename to `renderDeleteModal`? 😊